Makefile

```diff
@@ -88,51 +85,54 @@ NO_RE2C_BISON_PREFIX := -no-re2c-bison
 NO_TESTS_PREFIX := -no-tests
 
 ifeq ($(USE_NINJA),1)
-BUILD := ninja
+BUILD_CMD := ninja
 BUILD_FILE := build.ninja
 GENERATOR := Ninja
 else
-BUILD := $(MAKE) --no-print-directory
+BUILD_CMD := $(MAKE) --no-print-directory
 BUILD_FILE := Makefile
 GENERATOR := "Unix Makefiles"
 endif
 
 CMAKE_DIR = out/$($(1)_DIR)$($(2)_DIR)$($(3)_DIR)
-EXE_TARGET = $($(1)_PREFIX)$($(2)_PREFIX)$($(3)_PREFIX)
+BUILD_TARGET = $($(1)_PREFIX)$($(2)_PREFIX)$($(3)_PREFIX)
+INSTALL_TARGET = install-$($(1)_PREFIX)$($(2)_PREFIX)$($(3)_PREFIX)
 TEST_TARGET = test-$($(1)_PREFIX)$($(2)_PREFIX)$($(3)_PREFIX)
 
-define DEFAULT
-.PHONY: $(3)$($(4)_PREFIX) test$($(4)_PREFIX)
-$(3)$($(4)_PREFIX): $(call EXE_TARGET,$(1),$(2),$(4))
-	ln -sf ../$(call CMAKE_DIR,$(1),$(2),$(4))$(3) out/$(3)$($(4)_PREFIX)
-
-test$($(4)_PREFIX): $(call TEST_TARGET,$(1),$(2),$(4))
-test-everything: test$($(4)_PREFIX)
-endef
-
 define CMAKE
 $(call CMAKE_DIR,$(1),$(2),$(3)):
 	mkdir -p $(call CMAKE_DIR,$(1),$(2),$(3))
 
 $(call CMAKE_DIR,$(1),$(2),$(3))$$(BUILD_FILE): | $(call CMAKE_DIR,$(1),$(2),$(3))
 	cd $(call CMAKE_DIR,$(1),$(2),$(3)) && \
-	cmake -G $$(GENERATOR) $$(ROOT_DIR) $$($(1)_FLAG) $$($(2)_FLAG) $$($(3)_FLAG)
+	cmake -G $$(GENERATOR) -DCMAKE_INSTALL_PREFIX=$$(ROOT_DIR) $$(ROOT_DIR) $$($(1)_FLAG) $$($(2)_FLAG) $$($(3)_FLAG)
 endef
 
-define EXE
-.PHONY: $(call EXE_TARGET,$(1),$(2),$(3))
-$(call EXE_TARGET,$(1),$(2),$(3)): $(call CMAKE_DIR,$(1),$(2),$(3))$$(BUILD_FILE)
-	$$(BUILD) -C $(call CMAKE_DIR,$(1),$(2),$(3)) all
+define BUILD
+.PHONY: $(call BUILD_TARGET,$(1),$(2),$(3))
+$(call BUILD_TARGET,$(1),$(2),$(3)): $(call CMAKE_DIR,$(1),$(2),$(3))$$(BUILD_FILE)
+	$$(BUILD_CMD) -C $(call CMAKE_DIR,$(1),$(2),$(3)) all
+endef
+
+define INSTALL
+.PHONY: $(call INSTALL_TARGET,$(1),$(2),$(3))
+$(call INSTALL_TARGET,$(1),$(2),$(3)): $(call BUILD_TARGET,$(1),$(2),$(3))
+	$$(BUILD_CMD) -C $(call CMAKE_DIR,$(1),$(2),$(3)) install
 endef
 
 define TEST
 .PHONY: $(call TEST_TARGET,$(1),$(2),$(3))
 $(call TEST_TARGET,$(1),$(2),$(3)): $(call CMAKE_DIR,$(1),$(2),$(3))$$(BUILD_FILE)
-	$$(BUILD) -C $(call CMAKE_DIR,$(1),$(2),$(3)) run-tests
+	$$(BUILD_CMD) -C $(call CMAKE_DIR,$(1),$(2),$(3)) run-tests
+test-everything: $(CALL TEST_TARGET,$(1),$(2),$(3))
 endef
 
-.PHONY: all
-all: ${EXECUTABLES}
+.PHONY: all install
+all: install$(DEFAULT_SUFFIX)
```

Comment:
Why do you want install to be part of 'all'?  Why not leave install as a separate target?